src/core/lib/channel/channel_stack_builder.c

```diff
@@ -124,6 +124,24 @@ bool grpc_channel_stack_builder_move_prev(
   return true;
 }
 
+grpc_channel_stack_builder_iterator *grpc_channel_stack_builder_iterator_find(
+    grpc_channel_stack_builder *builder, const char *filter_name) {
+  GPR_ASSERT(filter_name != NULL);
+  grpc_channel_stack_builder_iterator *it =
+      grpc_channel_stack_builder_create_iterator_at_first(builder);
+  if (grpc_channel_stack_builder_iterator_is_end(it)) return it;
+  do {
+    const char *filter_name_at_it =
+        grpc_channel_stack_builder_iterator_filter_name(it);
+    if (filter_name_at_it != NULL &&
```

Comment:
In fact it'll always be NULL the first iteration. The first element of the list `it` iterates over (ie, `builder->begin`) is a market node with its filter field always set to NULL. We could in fact do a `next(it)` after calling `grpc_channel_stack_builder_create_iterator_at_first(builder)` but it looks ugly.